vignettes/adcoeq.Rmd

```diff
@@ -0,0 +1,215 @@
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+## Original Items {#original_items}
+
+The original items, i.e. the answers to the questionnaire questions, can be handled in the same way as in an [{admiral} BDS finding ADaM](https://pharmaverse.github.io/admiral/articles/bds_finding.html). 
+
+```{r eval=TRUE}
+adcoeq1 <- qs %>%
+  # Add ADSL variables
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, TRTEDT, TRT01P, TRT01A),
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # Add analysis parameter variables
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  ) %>%
+  # Add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = if_else(ADT <= TRTSDT, "BASELINE", VISIT),
+    AVISITN = if_else(ADT <= TRTSDT, 0, VISITNUM)
+  ) 
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq1, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT)
+)
+```
+
+If the numeric values of the answers in `QSSTRESN` values are not ready to be used for deriving
+scores and require transformation, it is recommended that `QSSTRESN` is kept in
+the ADaM dataset for traceability, and the transformed value is stored in
+`AVAL`, since that's what will be used for the score calculation.
+For the CoEQ item 6 has to be handled in such way as in the "Positive Mood" section high values are supposed to indicate a positive outcome (original direction of item 6: 0 = Not at all anxious = Positive, 100 = Extremely anxious = Negative):
+
+```{r eval=TRUE}
+adcoeq2 <- adcoeq1 %>%
+  # Add analysis value variables
+  mutate(
+    AVAL = if_else(PARAMCD == "COEQ06", 100-QSSTRESN, QSSTRESN),
+    AVALC = if_else(PARAMCD == "COEQ20", QSORRES, NA_character_)
+  ) 
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq2, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT, AVALC, AVAL),
+  filter = PARAMCD %in% c("COEQ01", "COEQ02", "COEQ03", "COEQ04", "COEQ05", "COEQ06", "COEQ07", "COEQ08", "COEQ09", "COEQ20")
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
+
+
+## Derive the four Subscales {#subscales}
+
+Scales and Scores are often derived as the sum or the average across a subset of
+the items. 
+For the Control of Eating Questionnaire, four subscales are derived:
+
+* Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
+
+* Craving for Sweet: Calculate average of items 3, 13, 14 and 15.
+
+* Craving for Savoury: Calculate average of items 4, 16, 17 and 18.
+
+* Positive Mood: Calculate average of items 5, 7, 8 and 6 (reversed).
```

Comment:
```suggestion
* Positive Mood: Calculate average of items 5, 7, 8 and 6 (inversed).
```

Reply:
Wouldn't inverse indicate "1/"? In which case reverse would be clearer?